pkg/build/builder/common.go

```diff
@@ -488,7 +489,7 @@ func replaceImagesFromSource(node *parser.Node, imageSources []buildapiv1.ImageS
 				names[stage.Name] = image
 			case child.Value == dockercmd.Copy:
 				if ref, ok := nodeHasFromRef(child); ok {
-					if len(ref) > 0 {
+					if ok && len(ref) > 0 {
```

Comment:
(I didn’t really read the new code — just skimming to see what is affected.

`ok` was checked one line above; this condition could be folded there.)

Reply:
Oof, right.